source/Irrlicht/CGLTFMeshFileLoader.cpp

```diff
@@ -206,17 +230,25 @@ core::vector2df CGLTFMeshFileLoader::MeshExtractor::readVec2DF(
 
 }
 
+/**
+ * Read a vector3df from a buffer at an offset.
+ * @return vec3 core::Vector3df
+*/
 core::vector3df CGLTFMeshFileLoader::MeshExtractor::readVec3DF(
 		const BufferOffset& readFrom,
-		const float scale)
+		const core::vector3df scale)
 {
 	return core::vector3df(
-		scale * readPrimitive<float>(readFrom),
-		scale * readPrimitive<float>(BufferOffset(readFrom, sizeof(float))),
-		-scale * readPrimitive<float>(BufferOffset(readFrom, 2 *
+		scale.X * readPrimitive<float>(readFrom),
+		scale.Y * readPrimitive<float>(BufferOffset(readFrom, sizeof(float))),
+		-scale.Z * readPrimitive<float>(BufferOffset(readFrom, 2 *

```

Comment:
This needs a unit test.

Reply:
Unfixed in 4a5c108b97ede77a72f1a5c2675006ba3df1c215